pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -154,12 +154,30 @@ type V2RegistriesConf struct {
 	Registries []Registry `toml:"registry"`
 	// An array of host[:port] (not prefix!) entries to use for resolving unqualified image references
 	UnqualifiedSearchRegistries []string `toml:"unqualified-search-registries"`
+	// ShortNameMode defines how short-name resolution should be handled by
+	// _consumers_ of this package.  Depending on the mode, the user should
+	// be prompted with a choice of using one of the unqualified-search
+	// registries when referring to a short name.
+	//
+	// Valid modes are:
+	//  * "default": prompt if stdout is a TTY
+	//  * "enforcing": always prompt and error if stdout is not a TTY
+	//  * "disabled", "": do not prompt and potentially use all unqualified-search registries
+	ShortNameMode string `toml:"short-name-mode"`
+
+	// shortNameMode is stored _once_ when loading the config.
+	shortNameMode types.ShortNameMode
```

Comment:
I’d prefer not mixing file formats and internal data.

Something like
```go
type parsedConfiguration struct {
  V2RegistriesConf // raw, merged - or only _some_ fields validly merged; or missing entirely
  shortNameMode
  parsedAliases 
}
```
could be generated and stored in the cache.

Reply:
Are you okay to add another TODO item (that I commit to addressing at a later point)? I think it makes sense to bundle this change together with breaking the big monolith of config loading into smaller pieces.

Reply:
Yes, this + the `TryReloadingConfig` return value + merging are all related.